======================================
Nextflow Fusion V5 pipeline handover
======================================

-----------------
Purpose
-----------------

This document records pipeline description, installation and execution

----

-----------------
Repo
-----------------

- `Nextflow Fusion V5 pipeline <https://github.com/ACTGenomics/hybridcapture_fusion_pipeline_nextflow>`_
- `Docker image construction for Fusion V4-based nextflow pipeline <https://github.com/ACTGenomics/Torrent_FUSION_calling.git>`_
- `Docker image construction for Arriba-based nextflow pipeline <https://github.com/ACTGenomics/fusion_pipeline_env>`_

-----------------
Docker Image
-----------------

Stable version for Fusion V4-based pipeline: `v0.23.0 <https://hub.docker.com/repository/docker/actgenomics/torrent_fusion_pipeline/general>`_
Stable version for Arriba-based pipeline: `v0.5 <https://hub.docker.com/repository/docker/actgenomics/fusion_dev/general>`_

-----------------
Others materials
-----------------

- Dev. environment: `Fusion dev. environment development <https://actg.atlassian.net/browse/ABIE-977>; Dev. image sumamry <https://actg.atlassian.net/browse/ABIE-993>`_
- Comparison study - Caller evalutaion: `Workflow comparison study <https://actg.atlassian.net/browse/ABIE-907>`_
- Design document - Fusion V4: `20240109_actfusion_v4_amplicon_workflow_ST.v0.28.0.pptx <https://actgenomics-my.sharepoint.com/:p:/p/yufenghuang/ETgxdTAFgUNLpYgrhERNdbMBU8HbGUqji5q6WKK-UmKrlQ?e=coyvwP>`_
- Dev. documents and records - Fusion V4: `V4_Final <https://actgenomics-my.sharepoint.com/:f:/p/yufenghuang/Eif-bxfsRuhNux2LYcluv7IBs0WAkjKUioDo8mivXrKljQ?e=bCIrwt>`_

----

-----------------
Files
-----------------

These refers to the files found in the `hybridcapture_fusion_pipeline_nextflow` repo

- **Workflow files**: 
    - ``hybridcapture_fusion_pipeline_nextflow/Workflow_docker_arriba.nf``
    - ``hybridcapture_fusion_pipeline_nextflow/Workflow_docker_illumina.nf``
- **Modules**: 
    - ``hybridcapture_fusion_pipeline_nextflow/modules/modules.nf``
    - ``hybridcapture_fusion_pipeline_nextflow/modules/k8s_modules.nf``
    - ``hybridcapture_fusion_pipeline_nextflow/modules/SampleInfo_k8s_illumina.nf``
- **Configs**:
    - ``hybridcapture_fusion_pipeline_nextflow/dockerconfigs/arriba_localdocker.config``
    - ``hybridcapture_fusion_pipeline_nextflow/dockerconfigs/fusion_multi_localdocker.v9.20241125.v0.23.0.config``
- **Test params files**: 
    - ``hybridcapture_fusion_pipeline_nextflow/params/IVTALL-1_arriba_test.json``
    - ``hybridcapture_fusion_pipeline_nextflow/params/fus_illumina_fastq.json``

Processes
~~~~~~~~~~~~~~

These processes are components of the Fusion V5 Fusion V4-based pipeline 

- **mergefastq**: Merge the two input paired-end fastq files via fastp
- **trimadap**: Perform sequencing adapter trimming on the merged fastq file by trimadap
- **fastp**: Perform sequencing reads filtering - discard reads with poor sequencing quality via fastp
- **bwa_isoform**: Perform isoform alignment for each merged read against the fusion target genes - this process is aimed to remove the reads that are highly similar to the sequences stored in the isoform database and can be skipped by adjusting the 'isoformfilteringflag' to 0 in the pipeline configuration file 
- **bwa_se**: Perform preferred transcriptome, preferred transcriptome with 10*N as intron sequences, alignment for each read 
- **fusioncalling**: Classify each aligned read generated by the bwa_se process to different fusion types
- **fuscall2QC_illumina**: Perform calling summary on the _callingresult.txt file generated by the fusioncalling process - output fusion summary file for clinical report {sampleName}_fusioncalling.boundary.QC.txt - will be stored in the 'Report' folder

These processes are components of the Fusion V5 Arriba-based pipeline 

- **STAR_ALIGN**: Generates STAR alignments for the input paired-end reads - the STAR paramters are recommended by the arriba caller
- **ARRIBA_CALL**: Perform fusion calling - parse the {sampleName}.Aligned.out.bam file and generate the two output files - {sampleName}.fusions.tsv & {sampleName}.fusions.discarded.tsv
- **SORT_INDEX_BAM**: Sort the STAR alignments and index the coordinate-sorted bam - alignments preprocessing for further analysis

----

--------------------
Prepare params file
--------------------

A custom script generate different types of params file, given the UUID and workflow type. 
A custom script to generate fusionv5.params.json, given the sample.info.csv, fastq directory, pu

.. note::

    The script generate paths from production volumes, therefore the samples should already been processed by legacy pipeline.


.. code-block:: console

    python3 /mnt/RD_Develop/sandyteng/ACTFusionV5/code/illumina_samplefilelist_csv.py \
        --csv "/mnt/RD_Develop/sandyteng/ACTFusionV5/nextflow/sample_info_AD/250509_NB552005_0507 ACTFusion v5 Probe feasibility test 2 condition.ST.csv" \
        --fastq_dir /mnt/RD_Result/2025-05-12_ABIE-1022/BCL2FASTQ/AANB01_507/ \
        --publish_dir /mnt/RD_Develop/sandyteng/ACTFusionV5/nextflow_outdir/AANB01_507/ \
        --json_out fusionv5.params.json \
        --runname AANB01_507

-----

--------------------
Configs fields
--------------------

The config fields for the Fusion V4-based workflow

- **refFile**: Fasta of the preferred transcripts (pseudo 10*N => intron) (MANE v0.95 + GENCODE-r38)
- **ambFile**: bwa index file (.amb file) derived from refFile 
- **annFile**: bwa index file (.ann file) derived from refFile 
- **bwtFile**: bwa index file (.bwt file) derived from refFile 
- **pacFile**: bwa index file (.pac file) derived from refFile 
- **saFile**: bwa index file (.sa file) derived from refFile 
- **annoFile**: The annotation file derived from refFile and primerlabelFile
- **gannoFile**: The annotation file derived from GENCODE-r38
- **isoformfaFile**: Fasta of the isoforms of the 26 target genes (RefSeq + GENCODE-r38)
- **isoformambFile**: bwa index file (.amb file) derived from isoformfaFile
- **isoformannFile**: bwa index file (.ann file) derived from isoformfaFile
- **isoformbwtFile**: bwa index file (.bwt file) derived from isoformfaFile
- **isoformpacFile**: bwa index file (.pac file) derived from isoformfaFile
- **isoformsaFile**: bwa index file (.sa file) derived from isoformfaFile
- **isoformmetaFile**: The annotation file derived from isoformfaFile
- **isoformfilteringflag**: isoform filtering step switch (1: enable filtering, 0: disable filtering) (process "bwaisoform")
- **truncatedmode**: truncated mode for funcational count summary (process "fuscall2QC_illumina")
- **truncatedseq_min_aligned_len**: minimum aligned length for the truncated sequence (default = 12 a.a.) (process "fuscall2QC")
- **inSpikeinFastqR1**: spike-in sequence to prevent pipeline termination (process "mergefastq"/"bam2fastq")
- **pdbFile**: 26 protein sequences (fasta) of the corresponding target transcripts
- **pdbmFile**: The ENST ID to UniProt ID map for pdbFile
- **adapFile**: adapter sequence to trim (for universal primer removal) (process "trimadap")
- **qcconfigFile**: Adjustable QC settings (default settings designed for amplicon based assay)
- **readqcconfigFile**: Adjustable QC settings (default settings designed for amplicon based assay)
- **primerlabelFile**: the designed primer region and the corresponding meta data
- **incqctemplateFile**: Adjustable QC settings (default settings designed for amplicon based assay)
- **boundaryqcFile**: Adjustable QC settings (default settings designed for amplicon based assay)
- **fusion_container**: latest pipeline image (actgenomics/torrent_fusion_pipeline:v0.23.0)

The config fields for the Arriba-based workflow

- **fusion_container**: latest pipeline image (actgenomics/fusion_dev:v0.5)
- **star_index_dir**: the folder containing pregenerated STAR index files (GRCh38+RefSeq)
- **annotation_gtf**: the gtf file for fusion breakpoint annotation (GRCh38+RefSeq)
- **assembly_fa**: the genome sequence provided by arriba's repo (GRCh38)
- **blacklist_tsv**: the blacklist curated by arriba (GRCh38)
- **known_fusions_tsv**: the known fusions curated by arriba (GRCh38)
- **protein_domains_gff3**: the protein domain file curated by arriba (GRCh38)

Remark:
- Reference files download (Arriba-based workflow)

.. code-block:: console

    # public image for arriba workflow (https://hub.docker.com/r/uhrigs/arriba/tags)
    docker pull uhrigs/arriba:2.4.0
    
    # download reference "GRCh38+RefSeq" (local directory: /mnt/RD_Develop/sandyteng/FusionCaptureTools/refdb_arriba/)
    docker run --rm -v /mnt/RD_Develop/sandyteng/FusionCaptureTools/refdb_arriba/:/references uhrigs/arriba:2.4.0 download_references.sh GRCh38+RefSeq

----

--------------------
Execution
--------------------

Server should already have native nextflow installed, if not uses conda to create a Nextflow run environment

.. code-block:: console

    ### The following commands are copied from the latest sequencing run "AANB01_507" (ref. issue: https://actg.atlassian.net/browse/ABIE-1024)

    # Execute Fusion V4-based pipeline with v0.95 db files (same db files for the amplicon-based workflow)
    nextflow run /mnt/RD_Develop/sandyteng/ACTFusionV5/repo_dev/hybridcapture_fusion_pipeline_nextflow/Workflow_docker_illumina.nf \
        -params-file /mnt/RD_Develop/sandyteng/ACTFusionV5/nextflow/testparams/param.ACTFusion_V5_AANB01_507-fusionv4-based.v0.1.json \
        -c /mnt/RD_Develop/sandyteng/ACTFusionV5/repo_dev/hybridcapture_fusion_pipeline_nextflow/dockerconfigs/fusion_multi_localdocker.v9.20241125.v0.23.0.config
    
    # Execute Fusion V4-based pipeline with v1.4 db files (db-v3.1, see issue: https://actg.atlassian.net/browse/ABIE-1012 for db construction)
    nextflow run /mnt/RD_Develop/sandyteng/ACTFusionV5/repo_dev/hybridcapture_fusion_pipeline_nextflow/Workflow_docker_illumina.nf \
        -params-file /mnt/RD_Develop/sandyteng/ACTFusionV5/nextflow/testparams/param.ACTFusion_V5_AANB01_507-fusionv4-based.v0.1-db-v3.1.json \
        -c /mnt/RD_Develop/sandyteng/ACTFusionV5/nextflow/repo_code_v1.4_dbtest_0414.2025/dockerconfigs/fusion_multi_localdocker.v9.20241125.v0.23.0_v1.4.MANE.transcriptome.v3-1.config

    # Execute Arriba-based pipeline 
    nextflow run /mnt/RD_Develop/sandyteng/ACTFusionV5/repo_dev/hybridcapture_fusion_pipeline_nextflow/Workflow_docker_arriba.nf \
        -params-file /mnt/RD_Develop/sandyteng/ACTFusionV5/nextflow/testparams/param.ACTFusion_V5_AANB01_507-arriba-based.v0.1.json \
        -c /mnt/RD_Develop/sandyteng/ACTFusionV5/repo_dev/hybridcapture_fusion_pipeline_nextflow/dockerconfigs/arriba_localdocker.config 

--------------------
Conclusion
--------------------

This completes the instructions for running Fusion V5 pipelines.